src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -1956,6 +1957,9 @@ void Unit::AttackerStateUpdate (Unit* victim, WeaponAttackType attType, bool ext
     if (attType != BASE_ATTACK && attType != OFF_ATTACK)
         return;                                             // ignore ranged case
 
+    if (!IsControlledByPlayer())
```

Comment:
m_ControlledByPlayer is set on any charm type, i guess u should use ispossessed().